File: scrimage-filters/src/test/scala/com/sksamuel/scrimage/filter/BlurFilterTest.scala
package com.sksamuel.scrimage.filter

import com.sksamuel.scrimage.ImmutableImage
import com.sksamuel.scrimage.nio.PngWriter
import org.scalatest.FunSuite

class BlurFilterTest extends FunSuite {

  implicit val writer: PngWriter = PngWriter.MaxCompression
  private val folder = "/com/sksamuel/scrimage/filters/blur"

  test("filter output matches expected") {
    assert(ImmutableImage.fromResource("/bird_small.png").filter(new BlurFilter) == ImmutableImage.fromResource(s"$folder/bird_small_blur.png"))
    assert(ImmutableImage.fromResource("/caviar.jpg").filter(new BlurFilter) == ImmutableImage.fromResource(s"$folder/caviar_blur.png"))
    assert(ImmutableImage.fromResource("/love.jpg").filter(new BlurFilter) == ImmutableImage.fromResource(s"$folder/love_blur.png"))
    assert(ImmutableImage.fromResource("/masks.jpg").filter(new BlurFilter) == ImmutableImage.fromResource(s"$folder/masks_blur.png"))
  }
}

File: scrimage-filters/src/test/kotlin/com/sksamuel/scrimage/filter/BlurFilterTest.kt
@file:Suppress("BlockingMethodInNonBlockingContext")

package com.sksamuel.scrimage.filter

import com.sksamuel.scrimage.ImmutableImage
import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe

class BlurFilterTest : FunSpec({

   val folder = "/com/sksamuel/scrimage/filters/blur"

   test("filter output matches expected") {
      ImmutableImage.fromResource("/bird_small.png").filter(BlurFilter()) shouldBe
         ImmutableImage.fromResource("$folder/bird_small_blur.png")

      ImmutableImage.fromResource("/caviar.jpg").filter(BlurFilter()) shouldBe
         ImmutableImage.fromResource("$folder/caviar_blur.png")

      ImmutableImage.fromResource("/love.jpg").filter(BlurFilter()) shouldBe
         ImmutableImage.fromResource("$folder/love_blur.png")

      ImmutableImage.fromResource("/masks.jpg").filter(BlurFilter()) shouldBe
         ImmutableImage.fromResource("$folder/masks_blur.png")
   }

})
